ejabberd_bridge/management/commands/ejabberd_auth.py

```diff
@@ -47,33 +49,37 @@ def to_ejabberd(self, answer=False):
         sys.stdout.flush()
 
     def auth(self, username=None, server="localhost", password=None):
-        self.logger.debug("Authenticating %s with password %s on server %s" % (username, password, server))
-        #TODO: would be nice if this could take server into account
-        user = authenticate(username=username, password=password)
-        return user and user.is_active
+        self.logger.debug("Authenticating %s on server %s" % (username, server))
+
+        try:
+            user_obj = get_user_model().objects.get(id=username)            
+            user = authenticate(username=user_obj.username, password=password)
+
+            if user :
+                return True
+            else:
+                return False
+        except User.DoesNotExist:
+            return False
 
     def isuser(self, username=None, server="localhost"):
         """
         Checks if the user exists and is active
         """
         self.logger.debug("Validating %s on server %s" % (username, server))
-        #TODO: would be nice if this could take server into account
+
         try:
             user = get_user_model().objects.get(username=username)
-            if user.is_active:
-                return True
-            else:
-                self.logger.warning("User %s is disabled" % username)
-                return False
+            return True
         except User.DoesNotExist:
             return False
 
     def setpass(self, username=None, server="localhost", password=None):
```

Comment:
we probably don't want to allow user to change their password from XMPP server

Reply:
agreed